modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/ElasticsearchIndexSearcher.java

```diff
@@ -117,41 +128,15 @@ else if (end < 0) {
 
 			Hits hits = null;
 
-			while (true) {
-				SearchSearchRequest searchSearchRequest =
-					createSearchSearchRequest(
-						searchRequest, searchContext, query, start, end);
-
-				SearchSearchResponse searchSearchResponse =
-					_searchEngineAdapter.execute(searchSearchRequest);
-
-				if (_log.isInfoEnabled()) {
-					_log.info(
-						StringBundler.concat(
-							"The search engine processed ",
-							searchSearchResponse.getSearchRequestString(),
-							" in ", searchSearchResponse.getExecutionTime(),
-							" ms"));
-				}
-
-				_populateResponse(searchSearchResponse, searchResponseBuilder);
-
-				searchResponseBuilder.searchHits(
-					searchSearchResponse.getSearchHits());
-
-				hits = searchSearchResponse.getHits();
-
-				Document[] documents = hits.getDocs();
-
-				if ((documents.length != 0) || (start == 0)) {
-					break;
-				}
-
-				int[] startAndEnd = SearchPaginationUtil.calculateStartAndEnd(
-					start, end, hits.getLength());
-
-				start = startAndEnd[0];
-				end = startAndEnd[1];
+			if (FeatureFlagManagerUtil.isEnabled("LPS-172416")) {
```

Comment:
I think we should test here whether (end > maxResultWindow) and just enter the default _search if not.

Reply:
Already tested it but we cant do that because we need to use the same sort